blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx

```diff
@@ -34,13 +50,25 @@ class RunDetailsHeader extends Component {
         // FIXME-FLOW: Remove the ":any" cast after completion of https://github.com/facebook/flow/issues/1059
         const changeSet = run.changeSet;
         const status = run.getComputedResult();
-        const durationMillis = run.isRunning() ?
-            moment().diff(moment(run.startTime)) : run.durationInMillis;
+        // we need to make sure that we calculate with the correct time offset
+        const skewMillis = this.context.config.getServerBrowserTimeSkewMillis();
+        // the time when we started the run harmonized with offset
+        const isRunning = () => run.isRunning() || run.isPaused();
+        const {
+            durationMillis,
+            endTime,
+            startTime,
+        } = timeManager.harmonizeTimes({
+            endTime: run.endTime,
+            startTime: run.startTime,
+            durationInMillis: run.durationInMillis,
+        }, skewMillis);
```

Comment:
And (NIT :) ) I don't get the benefit of the destruction here. How is all that more clear than a simple ...

```javascript
const harmonizedTimes = timeManager.harmonizeTimes(run, skewMillis);
logger.debug('rundetails times:', harmonizedTimes);
```